Updated Docstrings and function arguments, added docstring examples

Sandstorm831 opened the pull request:
closes #1406

chie2727 commented on ivy/functional/ivy/linear_algebra.py:
```diff
@@ -342,14 +345,31 @@ def diagonal(
     ivy.array([1.])
 
     A 3-D Example
```
Hi @Sandstorm831 , the edits are looking good so far.
You still need to rename `A 3-D Example` to `Functional Examples`, include the remaining required examples in this section, and also include an `Instance Method Examples` section.

Make sure to refer to the Docstring guide to see exactly what time of examples you need to include:
https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html

Also, have a look at the completed examples for the function add() to get an idea of how the overall formatting should be:
https://github.com/unifyai/ivy/blob/7d510b6cbc71350c6fa22d8273ba5326424eb655/ivy/functional/ivy/elementwise.py#L124

Sandstorm831 replied:
> Hi @Sandstorm831 , the edits are looking good so far. You still need to rename `A 3-D Example` to `Functional Examples`, include the remaining required examples in this section, and also include an `Instance Method Examples` section.
> 
> Make sure to refer to the Docstring guide to see exactly what time of examples you need to include: https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html
> 
> Also, have a look at the completed examples for the function add() to get an idea of how the overall formatting should be:
> 
> https://github.com/unifyai/ivy/blob/7d510b6cbc71350c6fa22d8273ba5326424eb655/ivy/functional/ivy/elementwise.py#L124

Hello,
Thank You for you suggestions, I have added examples of Instances methods, renamed to functional examples and also added more docstring examples.

Sandstorm831 commented:
https://github.com/unifyai/ivy/pull/1444#discussion_r897612611
Hello,
Thank You for you suggestions, I have added examples of Instances methods, renamed to functional examples and also added more docstring examples.

Sandstorm831 commented:
Hello,
Thank You for you suggestions, I have added examples of Instances methods, renamed to functional examples and also added more docstring examples.

chie2727 commented on ivy/functional/ivy/linear_algebra.py:
```diff
@@ -340,16 +343,124 @@ def diagonal(
+               [10, 14],
+               [19, 23]], dtype=int32)
+
+    Changing the axis arguments:
```
You can delete this line as it is not necessary

chie2727 replied:
You can delete line 387 as it is not necessary

chie2727 commented on ivy/functional/ivy/linear_algebra.py:
```diff
@@ -340,16 +343,124 @@ def diagonal(
+               [5.,8.]])
 
-    A 3-D Example
+    Functional examples
```
Thanks for making the edits! Just a few more changes to make:

- [ ] 1. Move the example in lines 346~353 into the `Functional examples` section because it is also classified as a functional example

- [ ] 2. Insert the `-----------` line under `Function examples`, and please change the `e` -> `E` as we need to keep the formatting consistent.

- [ ] 3. Insert a line-space between each example.

- [ ] 4. I've noted two lines that you should delete so please delete those.

- [ ] 5.  Include a functional example that takes an `ivy.Container` input

- [ ] 6.  Include an instance method example that takes an `ivy.Container` instance method

Sandstorm831 replied:
> Thanks for making the edits! Just a few more changes to make:
> 
> * [ ]  1. Move the example in lines 346~353 into the `Functional examples` section because it is also classified as a functional example
> * [ ]  2. Insert the `-----------` line under `Function examples`, and please change the `e` -> `E` as we need to keep the formatting consistent.
> * [ ]  3. Insert a line-space between each example.
> * [ ]  4. I've noted two lines that you should delete so please delete those.
> * [ ]  5.  Include a functional example that takes an `ivy.Container` input
> * [ ]  6.  Include an instance method example that takes an `ivy.Container` instance method

Hello @chie2727, thanks for the valuable suggestions, I have made the necessary changes please let me know if anything else has to be changed.

Sandstorm831 replied:
> Thanks for making the edits! Just a few more changes to make:
> 
> * [ ]  1. Move the example in lines 346~353 into the `Functional examples` section because it is also classified as a functional example
> * [ ]  2. Insert the `-----------` line under `Function examples`, and please change the `e` -> `E` as we need to keep the formatting consistent.
> * [ ]  3. Insert a line-space between each example.
> * [ ]  4. I've noted two lines that you should delete so please delete those.
> * [ ]  5.  Include a functional example that takes an `ivy.Container` input
> * [ ]  6.  Include an instance method example that takes an `ivy.Container` instance method

I am trying to tab on button for re review request but, it is not working, so commenting here. @chie2727

Sandstorm831 commented:
Hello @chie2727, thanks for the valuable suggestions, I have made the necessary changes please let me know if anything else has to be changed.

Sandstorm831 commented:
> Hello @chie2727, thanks for the valuable suggestions, I have made the necessary changes please let me know if anything else has to be changed.

Hello @chie2727  can you please give a review here, as I am unable to request re review.

chie2727 commented on ivy/functional/ivy/linear_algebra.py:
```diff
@@ -337,19 +340,163 @@ def diagonal(
     ivy.array([1.])
 
-    A 3-D Example
+    Functional Examples
```
Hi @Sandstorm831 , thanks for making all the changes!

It seems you're still yet to move the examples above (Lines 338~346) down into `Functional Examples`. \
Please move these 2 examples into `Function Examples` and then delete the heading `Examples`.

Once you make this change I'll merge in all your edits.

Sandstorm831 replied:
Hello @chie2727 , Thanks for giving the final change prompts, I have implemented all the required changes, please look through them once.

Sandstorm831 commented:
Hie @chie2727 , I have implemented all the required changes.

chie2727 commented on ivy/functional/ivy/linear_algebra.py:
```diff
@@ -380,26 +383,173 @@ def diagonal(
+    ivy.array([[1., 4.],
+               [5., 8.]])
+
+    >>> y = ivy.arange(9).reshape((3, 3))
```
Thanks for all the changes!

I was doing one look-through and I noticed that for some of your examples - particularly the ones that start with `ivy.arange().reshape()` - you haven't actually used an `ivy.array` input or `ivy.native_array` input in these.

chie2727 commented on ivy/functional/ivy/linear_algebra.py:
```diff
@@ -380,26 +383,173 @@ def diagonal(
+               [3, 4, 5],
+               [6, 7, 8]], dtype=int32)
+
+    >>> d = ivy.diagonal(y, -1)
```
And for some of these you need to specify again the full array `y` that is being used at the start of the example.

Please go through the examples again to make changes where necessary and let me know when you have made the edits!

Sandstorm831 commented:
Hello @chie2727, Thanks for suggesting changes. I have made some changes best of my understanding, please check them once.

chie2727 commented:
Instance methods and ivy.Container inputs actually don't work for this function so apologies for having requested you include these.
I've removed the examples that aren't necessary now so I can merge in the rest of your changes.

Sandstorm831 commented:
Hi @chie2727, Thanks for the review, I am facing the issue, as, on my local machine, the instance method and container are not working, I have also asked the same on discord, but didn't get any reply, and thanks for merging.
